src/node_jsvmapi.cc

```diff
@@ -1289,7 +1290,7 @@ napi_status napi_throw(napi_env e, napi_value error) {
       v8impl::V8LocalValueFromJsValue(error));
   // any VM calls after this point and before returning
   // to the javascript invoker will fail
-  return napi_ok;
+  return GET_RETURN_STATUS();
```

Comment:
Yeah, good point. I agree this should just return a flat napi_ok from here. I guess we could check to see if the result of ThrowException is empty and return an error in that case.